pkg/loki/modules.go

```diff
@@ -1738,7 +1753,13 @@ func (t *Loki) initAnalytics() (services.Service, error) {
 		return nil, err
 	}
 
-	objectClient, err := storage.NewObjectClient(period.ObjectType, t.Cfg.StorageConfig, t.ClientMetrics)
+	var objectClient client.ObjectClient
+	if t.Cfg.StorageConfig.ThanosObjStore {
+		metrics := &client.Metrics{Registerer: prometheus.DefaultRegisterer}
+		objectClient, err = storage.NewObjectClientV2("analytics", period.ObjectType, t.Cfg.StorageConfig, metrics)
```

Comment:
loki already has a thanos specific ruler storage section [`t.Cfg.RulerStorage`](https://github.com/grafana/loki/pull/11713/files#diff-dfe68c6d3018ac004616d53414bdfc82d0bbb04eea91fdff7f63e0369b22b022R1055) as the code was taken from cortex project. Should we remove ruler specific changes from this? I can revive https://github.com/grafana/loki/pull/11713 and rebase changes from this branch to it